File: chap3/R/code1.r
## パッケージのインストール
library(tensorflow)
library(keras)

## データの取得と前処理
mnist = dataset_mnist()

# トレーニングデータ
x_train = mnist$train$x
x_train = array_reshape(x_train, c(nrow(x_train), 784)) / 255
y_train = to_categorical(mnist$train$y, 10)

# テストデータ
x_test = mnist$test$x
x_test = array_reshape(x_test, c(nrow(x_test), 784)) / 255
y_test = to_categorical(mnist$test$y, 10)

File: chap3/julia/code1.jl
using Flux
using Flux.Data.MNIST
using Flux: onehotbatch

## トレーニングデータ
xTrain = MNIST.images(:train)
xTrain = hcat(float.(vec.(xTrain))...)
yTrain = MNIST.labels(:train)
yTrain = onehotbatch(yTrain, 0:9)

## テストデータ
xTest = MNIST.images(:test)
xTest = hcat(float.(vec.(xTest))...)
yTest = MNIST.labels(:test)
yTest = onehotbatch(yTest, 0:9)